src/libraries/System.Runtime.Serialization.BinaryFormat/src/Utils/FormatterServices.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Runtime.Serialization.BinaryFormat;
+
+internal static class FormatterServices
+{
+    private static AssemblyNameInfo? s_coreLibAssemblyName;
+
+    internal static AssemblyNameInfo CoreLibAssemblyName => s_coreLibAssemblyName ??= AssemblyNameInfo.Parse(GetAssemblyNameIncludingTypeForwards(typeof(object)).AsSpan());
```

Comment:
```suggestion
    internal static AssemblyNameInfo CoreLibAssemblyName => s_coreLibAssemblyName ??= AssemblyNameInfo.Parse("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089");
```
Would it be better for this to be a hard-coded mscorlib name to make the behavior less coupled to the runtime that the payload reader is executed on?